apps/explorer/src/ui/utils/LinkWithQuery.tsx

```diff
@@ -29,22 +35,49 @@ export function useNavigateWithQuery() {
     return navigateWithQuery;
 }
 
-export const LinkWithQuery = forwardRef<HTMLAnchorElement, LinkProps>(
-    ({ to, ...props }) => {
-        const [toBaseURL, toSearchParamString] = (to as string).split('?');
+export function useSearchParamsMerged() {
+    const [searchParams, setSearchParams] = useSearchParams();
 
-        const toURLSearchParams = new URLSearchParams(toSearchParamString);
+    const setSearchParamsMerged = useCallback(
+        (
+            params: Record<string, any>,
+            navigateOptions?: Parameters<typeof setSearchParams>[1]
+        ) => {
+            const nextParams = new URLSearchParams(searchParams);
+            Object.entries(params).forEach(([key, value]) => {
+                if (typeof value === 'undefined' || value === null) {
+                    nextParams.delete(key);
+                } else {
+                    nextParams.set(key, value);
+                }
+            });
+            setSearchParams(nextParams, navigateOptions);
+        },
+        [searchParams, setSearchParams]
+    );
+
+    return [searchParams, setSearchParamsMerged] as const;
+}
 
+export const LinkWithQuery = forwardRef<HTMLAnchorElement, LinkProps>(
+    ({ to, ...props }) => {
+        const href = useHref(to);
         const [searchParams] = useSearchParams();
+        const [toBaseURL, toSearchParamString] = href.split('?');
 
-        const newParams = new URLSearchParams({
-            ...Object.fromEntries(searchParams),
-            ...Object.fromEntries(toURLSearchParams),
-        });
+        const mergedSearchParams = useMemo(() => {
+            return new URLSearchParams({
+                ...Object.fromEntries(searchParams),
```

Comment:
Probably this existed already, but don't we need to keep only the network param from the existing href?

For example here


https://user-images.githubusercontent.com/10210143/205625433-05f49dc2-d7dc-4370-ad95-71873d10d233.mov

module param is not necessary.

Reply:
Oh this is a good point, I'll update these helpers to only preserve a set of params.